### PR TITLE
Add multiple docsets spec

### PR DESCRIPTION
**Pull request by yufeih**
https://dev.azure.com/ceapex/Engineering/_workitems/edit/121405

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5102)

### CODE REVIEW

File: docs/designs/multiple-docsets.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  https://github.com/multidocset/test:
+  - files:
+      a/docfx.yml: 'name: a'
+      a/a.md:
+      b/folder/docfx.yml: 'name: b'
+      b/folder/b.md:
+outputs:
+  a/.publish.json: |
+    {
+        "name": "a",
```
Comment (fenxu) [inline docs/designs/multiple-docsets.md]: >a [](start = 17, length = 1)

If it's docset name, then I need to have another property "Porduct".

File: docs/designs/multiple-docsets.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        "files": [{
+            "source_path": "a.md",
```
Comment (fenxu) [inline docs/designs/multiple-docsets.md]: Change this value definition to path_to_repo_root

File: docs/designs/multiple-docsets.md (hunk 1 above)
Comment (yufeih) [reply docs/designs/multiple-docsets.md]: I'll redefine it as depot name

File: docs/designs/multiple-docsets.md (hunk 2 above)
Comment (yufeih) [reply docs/designs/multiple-docsets.md]: This remains as path to docset path as discussed.

**Pull request merged**